prisma/schema.prisma

```diff
@@ -23,6 +23,74 @@ datasource db {
+// A final achievement can consist of several under achievements
+model achievement_template {
+  id                            Int                       @id @default(autoincrement())
+  metrics                       String[]                  @default([])
+  templateFor                   template_for_enum
+  // Groups defines the "final-achievement"
+  group                         String                    @unique @db.VarChar
+  // Group order tells in which place a subordinate step / achievement is
+  groupOrder                    Int                      
+  // the final achievement is marked with isLast 
+  isLast                        Boolean
+  // what type of achievement is it
+  type                          achievement_type_enum
+  name                          String                    @db.VarChar
+  subtitle                      String                    @db.VarChar
+  description                   String                    @db.VarChar
+  image                         String                    @db.VarChar
+  // some achievements show actions that a user can/must perform next 
+  actionDescription             String?                   @db.VarChar
+  actionRedirectLink            String?                   @db.VarChar
+  actionType                    action_type_enum?
+  // defines what value must be reached to receive the achievement 
+  condition                     String                    @db.VarChar
+  // defined how values must be aggregated to evaluate the condition
+  conditionDataAggregations     Json                      @db.Json
+  isActive                      Boolean 
+}
+model user_achievement {
+  id                            Int                       @id @default(autoincrement())
+  metrics                       String[]                  @default([])
+  templateFor                   template_for_enum
+  group                         String                    @unique @db.VarChar
+  groupOrder                    Int                       
+  isLast                        Boolean
+  type                          achievement_type_enum
+  name                          String                    @db.VarChar
+  subtitle                      String                    @db.VarChar
+  description                   String                    @db.VarChar
+  image                         String                    @db.VarChar
+  actionDescription             String                    @db.VarChar
+  actionRedirectLink            String                    @db.VarChar
+  actionType                    action_type_enum
+  condition                     String                    @db.VarChar
+  conditionDataAggregations     Json                      @db.Json
+  isActive                      Boolean 
+  isSeen                        Boolean
+  isAchieved                    Boolean
+  achievedAt                    DateTime                  @db.Timestamp(6)
+}
+
+// Save (tracked) event as raw data in DB and evaluate at runtime
+model event {
```

Comment:
As we are in a global namespace, could we prefix with achievement_ ? Or is this really "the event"?

Reply:
it is the real - tracked - event (raw data)

Reply:
Yes but we only use it for the achievement system? So it's name should include that